File: 2021/day01.ts
import { readFileSync } from "fs";

let part1 = 0, part2 = 0;

const lines = readFileSync("input.txt", "utf8").trim().split("\n").map(l => +l);

for (let i = 1; i < lines.length; i++) {
  if (lines[i] > lines[i - 1]) part1++;
}

let prevSum = lines[0] + lines[1] + lines[2];
for (let i = 1; i < lines.length; i++) {
  let currentSum = lines[i] + lines[i + 1] + lines[i + 2];
  if (currentSum > prevSum) part2++;
  prevSum = currentSum;
}

console.log(part1);
console.log(part2);

File: 2021/day01.py
lines = list(map(int, open("input.txt").readlines()))

part1 = part2 = 0 

for i in range(len(lines)-1):
    if lines[i+1] > lines[i]:
        part1 += 1

prev_w = sum([lines[0], lines[1], lines[2]])
for i in range(len(lines)-2):
    w = sum([lines[i], lines[i+1], lines[i+2]])
    if w > prev_w:
        part2 += 1
    prev_w = w

print(part1)
print(part2)
